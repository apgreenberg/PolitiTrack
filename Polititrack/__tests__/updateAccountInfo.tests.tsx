import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import UpdateAccountInfocreen from '../src/view/screens/UpdateAccountInfoScreen';
import { useSelectorMock, userData } from './common';
import { toBeDisabled } from '@testing-library/jest-native';
expect.extend({ toBeDisabled });

/**
 * Tests the update account info page
 */
describe('Tests the updatr account page', () => {
  useSelectorMock.mockReturnValue(userData);
  test('Viewing the update account info page', () => {
    const { toJSON } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <UpdateAccountInfocreen />
      </MockedProvider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
  test('Test that submit is disabled when name/email input is empty', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <UpdateAccountInfocreen />
      </MockedProvider>
    );
    expect(getByTestId('nameSaveButton')).toBeDisabled();
  });
  test('Test that submit is enabled when name/email is not empty', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <UpdateAccountInfocreen />
      </MockedProvider>
    );
    fireEvent.changeText(getByTestId('nameInput'), 'Ari');
    expect(getByTestId('nameSaveButton')).not.toBeDisabled();
  });
  test('Test that submit is disabled when passwords is empty', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <UpdateAccountInfocreen />
      </MockedProvider>
    );
    expect(getByTestId('passwordSaveButton')).toBeDisabled();
  });
  test('Test that submit is disabled when passwords do not match', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <UpdateAccountInfocreen />
      </MockedProvider>
    );
    fireEvent.changeText(getByTestId('passwordInput1'), '123');
    fireEvent.changeText(getByTestId('passwordInput2'), '456');
    expect(getByTestId('passwordSaveButton')).toBeDisabled();
  });
  test('Test that submit is enabled when passwords match', () => {
    const { getByTestId } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <UpdateAccountInfocreen />
      </MockedProvider>
    );
    fireEvent.changeText(getByTestId('passwordInput1'), '123');
    fireEvent.changeText(getByTestId('passwordInput2'), '123');
    expect(getByTestId('passwordSaveButton')).not.toBeDisabled();
  });
});
