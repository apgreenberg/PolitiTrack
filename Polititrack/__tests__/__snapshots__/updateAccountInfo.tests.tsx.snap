// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the updatr account page Viewing the update account info page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomWidth": 3,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "textAlign": "center",
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        Name
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingTop": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="default name"
        rejectResponderTermination={true}
        style={
          Object {
            "borderWidth": 2,
            "flexGrow": 1,
            "fontSize": 20,
            "height": 40,
          }
        }
        testID="nameInput"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="nameSaveButton"
        >
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              disabled={true}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#cdcdcd",
                  },
                ]
              }
            >
              Save
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomWidth": 3,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "textAlign": "center",
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        Email
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingTop": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="ari.greenberg99@gmail.com"
        rejectResponderTermination={true}
        style={
          Object {
            "borderWidth": 2,
            "flexGrow": 1,
            "fontSize": 20,
            "height": 40,
          }
        }
        testID="emailInput"
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Object {
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="emailSaveButton"
        >
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              disabled={true}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#cdcdcd",
                  },
                ]
              }
            >
              Save
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomWidth": 3,
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 24,
              "textAlign": "center",
            },
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        Password
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "space-between",
            "flexDirection": "column",
            "flexGrow": 1,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Array [
              Object {
                "borderWidth": 2,
                "flexGrow": 1,
                "fontSize": 20,
                "height": 40,
              },
              Object {
                "marginBottom": 10,
              },
            ]
          }
          testID="passwordInput1"
          underlineColorAndroid="transparent"
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderWidth": 2,
              "flexGrow": 1,
              "fontSize": 20,
              "height": 40,
            }
          }
          testID="passwordInput2"
          underlineColorAndroid="transparent"
        />
        <Text>
          Passwords do not match.
        </Text>
      </View>
      <View
        style={
          Object {
            "flexShrink": 1,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="passwordSaveButton"
        >
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <Text
              disabled={true}
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "margin": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#cdcdcd",
                  },
                ]
              }
            >
              Save
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
